fix(#175): camt.054 `Ntfctn`>`CreDtTm` is optional

tlamy opened the pull request:
This PR fixes keeps the camt parser from throwing on optinal DateTime fields

Commit:
fix(#175): `Ntfctn`>`CreDtTm` is optional

PowerKiKi commented on src/Camt054/Decoder/Message.php:
```diff
@@ -19,10 +19,17 @@ public function addRecords(DTO\Message $message, SimpleXMLElement $document): vo
         foreach ($xmlNotifications as $xmlNotification) {
+            $recordDateStr = (string) $xmlNotification->CreDtTm;
+            if ($recordDateStr === '') {
+                $notificationDate = $fallbackCreationDate;
```
Can you quote the specification where it says that the mandatory `CreDtTm` on `GrpHdr` _must_ be used as a fallback for the optional `CreDtTm` on `Ntfctn` ?

tlamy replied:
I had to different AIs look this up for me, and the outcome was:
	•	There is no requirement that Ntfctn/CreDtTm must exist.
	•	If it is missing, then only GrpHdr/CreDtTm can be relied on.
But there is no specification that it _must_ be used, and I'm perfectly fine to have `null` returned from the getter instead.

PowerKiKi committed:
Leverage e2e test result as JSON

PowerKiKi reviewed (changes requested):
I simplified the tests for you. But before merging I would like to be sure that the fallback mechanism is indeed dictated by the specification. I suspect it we should not automate that mechanism inside this library, and instead let the end-user handle the situation as they see fit.

Commit:
fix(#175): `Ntfctn`>`CreDtTm` is optional; make nullable instead of using fallback

Commit:
fix(#175): reverted auto "cleanup" by ide

PowerKiKi committed:
Some `Record` have a required `createdOn`, some don't

Required `createdOn`:

- `Statement`
- `Report`

Optional `createdOn`:

- `Notification`

PowerKiKi committed:
Avoid code repetition

PowerKiKi commented:
I pushed commits too, because other classes must keep `createdOn` as required. Can you please confirm that it still works for you ?

tlamy commented:
(As long as the) Tests run fine, this is absolutely ok with me

PowerKiKi commented:
Released as https://github.com/genkgo/camt/releases/tag/2.10.3

Thank you for your contribution
